warehouse/accounts/models.py

```diff
@@ -237,6 +237,7 @@ def has_primary_verified_email(self):
     @property
     def recent_events(self):
         session = orm.object_session(self)
+        assert session is not None
```

Comment:
This (and the many other `assert session is not None`s makes me think we probably need a helper function that is responsible for narrowing the type and explicitly handling the case where the session might be `None` rather than assuming it always is and throwing an `AssertionError` if it's not. AFAICT, this probably should just raise our own exception with more details.

Reply:
I'll extract this to a helper and push as a separate PR, so that can be evaluated on its merits alone.

Reply:
https://github.com/pypi/warehouse/pull/17582